client/my-sites/checkout/src/hooks/use-get-thank-you-url/index.tsx

```diff
@@ -36,10 +36,13 @@ export default function useGetThankYouUrl( {
 	hideNudge,
 	isInModal,
 	domains,
+	connectAfterCheckout,
+	adminUrl: wpAdminUrl,
+	fromSiteSlug,
 }: GetThankYouUrlProps ): GetThankYouUrl {
 	const selectedSiteData = useSelector( ( state ) => getSelectedSite( state ) );
 
-	const adminUrl = selectedSiteData?.options?.admin_url;
+	const adminUrl = selectedSiteData?.options?.admin_url || wpAdminUrl;
```

Comment:
If `adminUrl` is defined here, does it need to be passed in props?
Also, does wpAdminUrl need to be collected from the state? I don't see that variable defined here 🤔

Reply:
> If adminUrl is defined here, does it need to be passed in props?

Yes, In the "connectAfterCheckout" flow, the site will not be in context (because it has not been connected to wordpress.com yet), and therefore attempting to retrieve the site's adminUrl from state will return `undefined` (again because there will be no site in context), so to remedy this we are getting the adminUrl from a checkout url query arg and passing it in as a prop here.
In other words, in the 'connectAfterCheckout' flow(site is not in context), the site's wp-admin url is passed in here as a prop, but in any other checkout flows where the site is in context, the site's wp-admin url is retrieved from state. 🙂

> Also, does wpAdminUrl need to be collected from the state? I don't see that variable defined here

No, in the 'connectAfterCheckout' flow the site's wp-admin url is passed into this component as prop `adminUrl` but renamed to `wpAdminUrl`. Then adminUrl is defined loosely as `const adminUrl = state.admin_url || props.wpAdminUrl`, if that makes sense..